lib/explorer/data_frame.ex

```diff
@@ -515,34 +556,82 @@ defmodule Explorer.DataFrame do
   @doc """
   Reads a parquet file into a dataframe.
 
+  It accepts a filename that can be a local file, a "s3://" schema, or
+  a FSS entry like `FSS.S3.Entry`.
+
   ## Options
 
     * `:max_rows` - Maximum number of lines to read. (default: `nil`)
 
     * `:columns` - A list of column names or indexes to keep. If present,
       only these columns are read into the dataframe. (default: `nil`)
+
+    * `:config` - An optional config struct or map, normally associated with the
+      entry provided. See [IO section](#module-io-operations) for more details. (default: `nil`)
+
   """
   @doc type: :io
-  @spec from_parquet(filename :: String.t(), opts :: Keyword.t()) ::
+  @spec from_parquet(filename :: String.t() | fs_entry(), opts :: Keyword.t()) ::
           {:ok, DataFrame.t()} | {:error, term()}
   def from_parquet(filename, opts \\ []) do
     {backend_opts, opts} = Keyword.split(opts, [:backend, :lazy])
 
     opts =
       Keyword.validate!(opts,
         max_rows: nil,
-        columns: nil
+        columns: nil,
+        config: nil
       )
 
     backend = backend_from_options!(backend_opts)
 
     backend.from_parquet(
-      filename,
+      normalise_entry(filename, opts[:config]),
       opts[:max_rows],
       to_columns_for_io(opts[:columns])
     )
   end
 
+  defp normalise_entry(%_{} = entry, config) when config != nil do
+    raise ArgumentError,
+          ":config key is only supported when the argument is a string, got #{inspect(entry)} with config #{inspect(config)}"
+  end
+
+  defp normalise_entry(%Local.Entry{} = entry, nil), do: entry
+  defp normalise_entry(%S3.Entry{config: %S3.Config{}} = entry, nil), do: entry
+
+  defp normalise_entry("s3://" <> _rest = entry, config) do
+    config = s3_config(config)
+    %S3.Entry{url: entry, config: config}
+  end
+
+  defp normalise_entry("file://" <> path, _config), do: %Local.Entry{path: path}
+
+  defp normalise_entry(filepath, _config) when is_binary(filepath) do
+    if File.exists?(filepath) do
+      %Local.Entry{path: filepath}
+    else
+      raise ArgumentError,
+            "cannot read entry because it's not a valid file, or its filesystem is not supported"
+    end
+  end
+
+  defp s3_config(%S3.Config{} = config), do: config
+
+  # should we support maps? and kw lists?
+  defp s3_config(%{access_key_id: _, secret_key_id: _} = config) do
+    %S3.Config{
+      region: config[:region],
+      access_key_id: config.access_key_id,
+      secret_access_key: config.secret_key_id
+    }
+  end
+
+  defp s3_config(other) do
+    raise ArgumentError,
+          "expected a valid S3 configuration struct or map, but instead found: #{inspect(other)}"
```

Comment:
Most people will call FSS.S3.config(...) or similar to build it once. In the future we could call it for them I guess but let's leave it a bit strict for now. :)

```suggestion
  defp s3_config(other) do
    raise ArgumentError, "expected a valid FSS.S3.config/1 in :config, got: #{inspect(other)}"
```

Reply:
Makes sense!